sources/sql_database/helpers.py

```diff
@@ -32,12 +32,13 @@
     TTypeAdapter,
 )
 
-from sqlalchemy import Table, create_engine
+from sqlalchemy import Table, create_engine, select
 from sqlalchemy.engine import Engine
 from sqlalchemy.exc import CompileError
 
 
 TableBackend = Literal["sqlalchemy", "pyarrow", "pandas", "connectorx"]
+TQueryAdapter = Callable[[SelectAny, Table], SelectAny]
```

Comment:
I'm thinking of extending this to `Callable[[SelectAny, Table, MetaData], SelectAny]`  
Including metadata will let you access other sqla tables and add joins and such.